qiskit/visualization/__init__.py

```diff
@@ -17,20 +17,160 @@
 
 .. currentmodule:: qiskit.visualization
 
+
+Overview
+========
+
+The visualization module contain functions that visualizes measurement outcome counts, quantum
+states, circuits, pulses, devices and more.
+
+Prerequisites
+=============
+
+To use visualization functions, you are required to install visualization optionals to your
+development environment:
+
+.. code-block:: bash
+
+   pip install 'qiskit[visualization]'
+
 Counts and State Visualizations
 ===============================
 
+This section contains functions that visualize measurement outcome counts and quantum states.
+
+APIs
+----
+
+Counts Visualization
+^^^^^^^^^^^^^^^^^^^^
+
 .. autosummary::
    :toctree: ../stubs/
 
    plot_histogram
+
+State Visualization
+^^^^^^^^^^^^^^^^^^^
+
+.. autosummary::
+   :toctree: ../stubs/
+
    plot_bloch_vector
    plot_bloch_multivector
    plot_state_city
    plot_state_hinton
    plot_state_paulivec
    plot_state_qsphere
 
+Example Usage
+-------------
+
+Counts Visualization
+^^^^^^^^^^^^^^^^^^^^
+
+Here is an example of using :func:`plot_histogram` to visualize measurement outcome counts:
+
+.. jupyter-execute::
+
+   from qiskit.visualization import plot_histogram
+
+   counts = {"00": 501, "11": 499}
+   plot_histogram(counts)
+
+The data can be a dictionary with bit string as key and counts as value, or more commonly a
+:class:`~qiskit.result.Counts` object obtained from :meth:`~qiskit.result.Result.get_counts`.
+
+State Visualization
+^^^^^^^^^^^^^^^^^^^
+
+Here is an example of using :func:`plot_state_city` to visualize a quantum state:
+
+.. jupyter-execute::
+
+   from qiskit.visualization import plot_state_city
+
+   state = [[ 0.75  , 0.433j],
+            [-0.433j, 0.25  ]]
+   plot_state_city(state)
+
+The state can be array-like list of lists, ``numpy.array``, or more commonly
+:class:`~qiskit.quantum_info.Statevector` or :class:`~qiskit.quantum_info.DensityMatrix` objects
+obtained from a :class:`~qiskit.circuit.QuantumCircuit`:
+
+.. jupyter-execute::
+
+   from qiskit import QuantumCircuit
+   from qiskit.quantum_info import Statevector, DensityMatrix
+
+   qc = QuantumCircuit(2)
+   qc.h(0)
+   qc.cx(0,1)
+
+   # plot using a Statevector
+   state = Statevector(qc)
+   plot_state_city(state)
+
+.. jupyter-execute::
+
+   # plot using a DensityMatrix
+   state = DensityMatrix(qc)
+   plot_state_city(state)
+
+You can find code examples for each visualization functions on the individual function `API page
+<#apis>`_.
+
+Common Keyword Arguments
+------------------------
+
+The figures created by counts and state visualizations functions are mostly generated by
+`Matplotlib <https://matplotlib.org/>`_. As a result there are keyargs inherited from
+``Matplotlib`` that are common to most functions, including:
```

Comment:
I'd pull this out into the top level of the file, perhaps - a lot of these arguments are common to functions beyond the counts and state visualizations (the device visualizations, for example), and it's fine to say something like

> Many of these figures are created by Matplotlib, and accept a subset of the following common arguments.  Consult the individual documentation for exact details.

Reply:
Good point! Made the suggested change.